src/pretix/base/timeframes.py

```diff
@@ -0,0 +1,388 @@
+        True,
+        pgettext_lazy('reporting_timeframe', 'by quarter'),
+        lambda start_d: f"Q{(start_d.month - 1) // 3 + 1}/{start_d.year}",
+    ),
+    (
+        'year_this',
+        pgettext_lazy('reporting_timeframe', 'Current year'),
+        lambda ref_d: ref_d.replace(day=1, month=1),
+        lambda ref_d, start_d: start_d.replace(day=31, month=12),
+        True,
+        pgettext_lazy('reporting_timeframe', 'by year'),
+        lambda start_d: str(start_d.year),
+    ),
+    (
+        'year_to_date',
+        pgettext_lazy('reporting_timeframe', 'Current year to date'),
+        lambda ref_d: ref_d.replace(day=1, month=1),
+        lambda ref_d, start_d: ref_d,
+        False,
+        pgettext_lazy('reporting_timeframe', 'by year'),
+        lambda start_d: str(start_d.year),
+    ),
+    (
+        'year_previous',
+        pgettext_lazy('reporting_timeframe', 'Previous year'),
+        lambda ref_d: (ref_d.replace(day=1, month=1) - timedelta(days=1)).replace(day=1, month=1),
+        lambda ref_d, start_d: start_d.replace(day=31, month=12),
+        False,
+        pgettext_lazy('reporting_timeframe', 'by year'),
+        lambda start_d: str(start_d.year),
+    ),
+    (
+        'year_next',
+        pgettext_lazy('reporting_timeframe', 'Next year'),
+        lambda ref_d: ref_d.replace(day=1, month=1, year=ref_d.year + 1),
+        lambda ref_d, start_d: start_d.replace(day=31, month=12),
+        True,
+        pgettext_lazy('reporting_timeframe', 'by year'),
+        lambda start_d: str(start_d.year),
+    ),
+)
+
+
+class DateFrameWidget(forms.MultiWidget):
+    template_name = 'pretixbase/forms/widgets/dateframe.html'
+
+    def __init__(self, *args, **kwargs):
+        self.timeframe_choices = kwargs.pop('timeframe_choices')
+        widgets = (
+            forms.Select(choices=self.timeframe_choices),
+            forms.DateInput(attrs={'class': 'datepickerfield', 'placeholder': pgettext_lazy('timeframe', 'Start')}),
+            forms.DateInput(attrs={'class': 'datepickerfield', 'placeholder': pgettext_lazy('timeframe', 'End')}),
+        )
+        super().__init__(widgets=widgets, *args, **kwargs)
+
+    def decompress(self, value):
+        if not value:
+            return ['unset', None, None]
+        if '/' in value:
+            return [
+                'custom',
+                date.fromisoformat(value.split('/', 1)[0]),
+                date.fromisoformat(value.split('/', 1)[-1]),
+            ]
+        return []
+
+    def get_context(self, name, value, attrs):
+        ctx = super().get_context(name, value, attrs)
+        ctx['required'] = self.timeframe_choices[0][0] == 'unset'
+        return ctx
+
+
+def _describe_timeframe(label, start, end, future, describe):
+    details = f'{date_format(start(now()), "SHORT_DATE_FORMAT")} - {date_format(end(now(), start(now())), "SHORT_DATE_FORMAT")}'
```

Comment:
Hm, I think this could be made more readable, especially when start and end are the same. What about using the existing daterange-helper? This IMHO creates more human-readable date ranges.

```suggestion
    from pretix.helpers.daterange import daterange # todo: probably import at top
    d_start = start(now())
    d_end = end(now(), d_start)
    details = daterange(d_start, d_end)
```

Generally some describe + date-range labels could be optimized, but that would mean a custom label-method for every date-range – although I think it could be possible creating readable describe-lambdas with `d_end` and the use of the daterange-helper.

Actually, we could make out lives even easier: I do not think at least all „by month“, „by quarter“ and „by year“ do not need any daterange-labels at all (so just the existing describe-lambda). Even the day-ranges probably do not need them. For the week-range I find it helpful though.

Reply:
I think for "last 7 days" it makes sense, and it also describes phrases like "quarter to date" to people who are not used to the phrase, even though it is quite common

Reply:
I changed it to be explicitly changeable for every option, what do you think?

Reply:
Perfect! Thanks!